src/Controller.php

```diff
@@ -34,7 +34,13 @@ private function getProvider($provider, $redirectUri = null)
     {
         $config = $this->getConfig($provider) + compact('redirectUri');
 
-        $class  = 'League\\OAuth2\\Client\\Provider\\' . ucfirst($provider);
+        if (isset($_ENV[strtoupper($provider) . '_NAMESPACE'])) {
+            $namespace = $_ENV[strtoupper($provider) . '_NAMESPACE'];
```

Comment:
Heh, I didn't think of that. Maybe instead of having this config, we could use the regex?